src/UglyToad.PdfPig/Content/OptionalContentGroupElement.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.Collections.Generic;
+using UglyToad.PdfPig.Tokens;
+
+namespace UglyToad.PdfPig.Content
+{
+    /// <summary>
+    /// An optional content group is a dictionary representing a collection of graphics
+    /// that can be made visible or invisible dynamically by users of viewers applications.
+    /// </summary>
+    public class OptionalContentGroupElement
+    {
+        /// <summary>
+        /// The type of PDF object that this dictionary describes.
+        /// <para>Must be OCG for an optional content group dictionary.</para>
+        /// </summary>
+        public string Type { get; }
+
+        /// <summary>
+        /// The name of the optional content group, suitable for presentation in a viewer application's user interface.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// A single name or an array containing any combination of names.
+        /// <para>Default value is 'View'.</para>
+        /// </summary>
+        public IReadOnlyList<string> Intent { get; }
+
+        /// <summary>
+        /// A usage dictionary describing the nature of the content controlled by the group.
+        /// </summary>
+        public IReadOnlyDictionary<string, IToken> Usage { get; }
+
+        /// <summary>
+        /// Underlying <see cref="MarkedContentElement"/>.
+        /// </summary>
+        public MarkedContentElement MarkedContent { get; }
+
+        internal OptionalContentGroupElement(MarkedContentElement markedContentElement)
+        {
+            MarkedContent = markedContentElement;
+
+            // Type - Required
```

Comment:
We might need to store an IPdfTokenScanner internally here and use the TryGetv overload that takes in the scanner since these can be indirect references I assume?

Reply:
will check that too